File: DyeHard/EnemyManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XNACS1Lib;
using Microsoft.Xna.Framework;


namespace Dyehard
{
    class EnemyManager
    {
        private const float enemyFrequency = 12f;
        private Hero hero;
        private List<Enemy> enemies;
        private Timer newEnemyTimer;

        public EnemyManager(Hero hero)
        {
            this.hero = hero;
            enemies = new List<Enemy>();
            newEnemyTimer = new Timer(enemyFrequency);
        }

        public void remove()
        {
            foreach (Enemy e in enemies) {
                e.remove();
            }
        }

        public void update()
        {
            // remove any dead enemies
            enemies.RemoveAll(enemy => !enemy.isAlive());

            foreach (Enemy currentEnemy in enemies)
            {
                currentEnemy.update();
            }

            // prevent generating enemy
            if (GameWorld.Speed != 0)
            {
                newEnemyTimer.update();
            }
            
            //generate new enemy
            if (newEnemyTimer.isDone())
            {
                float randomY = XNACS1Base.RandomFloat(GameWorld.topEdge - 5, GameWorld.bottomEdge + 5);
                Vector2 position = new Vector2(GameWorld.rightEdge + 5, randomY);
                
                switch (XNACS1Base.RandomInt(0,3))
                {
                    case 1:
                        enemies.Add(new BrainRobot(position, 3.25f, 6, hero));
                        break;
                    case 2:
                        enemies.Add(new WhiteRobot(position, 4.14f, 6, hero));
                        break;
                    default:
                        enemies.Add(new BlackRobot(position, 3.6f, 6, hero));
                        break;
                }

                newEnemyTimer.reset();
            }
        }

        public void draw()
        {
            foreach (Enemy b in enemies)
            {
                b.getPosition().TopOfAutoDrawSet();
            }
        }

        public void killAll()
        {
            enemies.Clear();
        }

        public List<Enemy> getEnemies()
        {
            return enemies;
        }
   
    }
}

File: Java/DyeHardAPI/src/Dyehard/Enemies/EnemyManager.java
package dyehard.Enemies;

import java.util.ArrayList;
import java.util.Random;

import Engine.Vector2;
import dyehard.UpdateManager;
import dyehard.Updateable;
import dyehard.Player.Hero;
import dyehard.Util.Timer;
import dyehard.World.GameWorld;

public class EnemyManager implements Updateable {
    // This time is in milliseconds
    private final float enemyFrequency = 10000f;
    private Hero hero;
    private ArrayList<Enemy> enemies;
    private Timer timer;

    public EnemyManager(Hero hero) {
        this.hero = hero;
        UpdateManager.register(this);
        enemies = new ArrayList<Enemy>();
        timer = new Timer(enemyFrequency);
    }

    @Override
    public void update() {
        // generate new enemy
        if (timer.isDone()) {
            Random rand = new Random();
            // TODO: Replace magic numbers
            float randomY = (GameWorld.TOP_EDGE - 5f - GameWorld.BOTTOM_EDGE + 5f)
                    * rand.nextFloat() + 0f + 5f;
            Vector2 position = new Vector2(GameWorld.RIGHT_EDGE + 5, randomY);
            switch (rand.nextInt(3)) {
            case 1:
                enemies.add(new BrainEnemy(position, 7.5f, hero));
                break;
            case 2:
                enemies.add(new RedBeamEnemy(position, 7.5f, hero));
                break;
            default:
                enemies.add(new SpiderEnemy(position, 7.5f, hero));
                break;
            }
            timer.reset();
        }
    }

    public ArrayList<Enemy> getEnemies() {
        return enemies;
    }

    @Override
    public boolean isActive() {
        // The enemy manager is always active
        return true;
    }
}
